datalad/support/dsconfig.py

```diff
@@ -0,0 +1,319 @@
+
+    #
+    # Compatibility with GitPython's ConfigParser
+    #
+    def get_value(self, section, option, default=None):
+        """Like `get()`, but with an optional default value
+
+        If the default is not None, the given default value will be returned in
+        case the option did not exist. This behavior immitates GitPython's
+        config parser.
+        """
+        try:
+            return self.get(section, option)
+        except KeyError as e:
+            # this strange dance is needed because gitpython does it this way
+            if default is not None:
+                return default
+            else:
+                raise e
+
+    #
+    # Modify configuration (proxy respective git-config call)
+    #
+    def _require_location(self, where, args=None):
+        if args is None:
+            args = []
+        if not where in ('dataset', 'local', 'global'):
+            raise ValueError(
+                "unknown configuration label '{}' (not 'dataset', or 'global')".format(
+                    where))
+        if where == 'dataset':
+            if not self._dataset:
+                raise ValueError(
+                    'ConfigManager cannot store to configuration to dataset, none specified')
+            # create an empty config file if none exists, `git config` will
+            # fail otherwise
+            dscfg_dirname = opj(self._dataset.path, '.datalad')
+            dscfg_fname = opj(dscfg_dirname, 'config')
+            if not exists(dscfg_dirname):
+                os.makedirs(dscfg_dirname)
+            if not exists(dscfg_fname):
+                open(dscfg_fname, 'w').close()
+            args.extend(['--file', opj(self._dataset.path, '.datalad', 'config')])
+        elif where == 'global':
+            args.append('--global')
+        elif where == 'local':
+            args.append('--local')
+        return args
+
+    def add(self, var, value, where='dataset', reload=True):
+        """Add a configuration variable and value
+
+        Parameters
+        ----------
+        var : str
+          Variable name including any section like `git config` expects them, e.g.
+          'core.editor'
+        value : str
+          Variable value
+        where : {'dataset', 'local', 'global'}, optional
+          Indicator which configuration file to modify. 'dataset' indicates the
+          persistent configuration in .datalad/config of a dataset; 'local'
+          the configuration of a dataset's Git repository in .git/config;
+          'global' refers to the general configuration that is not specific to
+          a single repository (usually in $USER/.gitconfig
+        reload : bool
+          Flag whether to reload the configuration from file(s) after
+          modification. This can be disable to make multiple sequential
+          modifications slightly more efficient.
+        """
+        args = self._require_location(where)
+        args += ('--add', var, value)
```

Comment:
holy cow -- I didn't know that you can += a tuple to a list... but still might be more kosher to have right side in a list for my sanity